cmake-tool/helpers/application_settings.cmake

```diff
@@ -96,89 +91,181 @@ endfunction()
 #
 # Calling this function will result in forced updates to the cache.
 function(correct_platform_strings)
-    set(_REWRITE ON)
+
+    if(KernelX86Sel4Arch)
+        # this used to be a common mechanism how x86 architectures were
+        # selected. It's deprecated now and PLATFORM should be used instead. As
+        # of Nov/2020, we don't make this an error to give everybody a chance
+        # to update their scripts.
+        if("${PLATFORM}" STREQUAL "")
+            if(NOT correct_platform_strings_no_print)
+                message(
+                    DEPRECATION
+                        "setting PLATFORM from deprecated KernelX86Sel4Arch: ${KernelX86Sel4Arch}"
+                )
+            endif()
+            set(PLATFORM "${KernelX86Sel4Arch}")
+        elseif("${PLATFORM}" STREQUAL "${KernelX86Sel4Arch}")
+            if(NOT correct_platform_strings_no_print)
+                message(DEPRECATION "KernelX86Sel4Arch is deprecated, use PLATFORM only")
+            endif()
+        else()
+            message(
+                FATAL_ERROR
+                    "PLATFORM=${PLATFORM} does not match KernelX86Sel4Arch=${KernelX86Sel4Arch}"
+            )
+        endif()
+    elseif(KernelArmSel4Arch)
+        # this has never been widely in use, so we stop supporting it
+        message(FATAL_ERROR "KernelArmSel4Arch is no longer supported, use PLATFROM")
+    elseif(KernelRiscVSel4Arch)
+        # this should not have been in use at all
+        message(FATAL_ERROR "KernelRiscVSel4Arch is no longer supported, use PLATFROM")
+    endif()
+
     set(
         correct_platform_strings_platform_aliases
-        sabre
-        wandq
-        kzm
-        rpi3
-        exynos5250
-        exynos5410
-        exynos5422
-        am335x-boneblack
-        am335x-boneblue
-        x86_64
-        ia32
+        # The elements of this list are:
+        #      "-"+<name of architecture #1 specific platform variable>
+        #      <chip family>,<board_1>,<board_2>...
+        #      <chip family>,<board_1>,<board_2>...
+        #      ...
+        #      "-"+<name of architecture #2 specific platform variable>
+        #      <chip family>,<board_1>,<board_2>...
+        #      ...
+        #  where this function will try to match PLATFORM against <board_n> and
+        #  then set:
+        #      KernelPlatform=<chip family>
+        #      <platform variable>=<board_n>
+        # Note that <board_n> must be a unique name, as the first match will be
+        # used. If there was no match for any board the function will set:
+        #     KernelPlatform=${PLATFORM}
+        # and leave all further setup to the architecture/platform specific
+        # configuration to <sel4 kernel>/src/plat/*/config.cmake
+        #
+        "-KernelARMPlatform"
+        "imx6,sabre,wandq"
+        "imx31,kzm"
+        "bcm2837,rpi3"
+        "exynos5,exynos5250,exynos5410,exynos5422"
+        "am335x,am335x-boneblack,am335x-boneblue"
+        "-KernelSel4Arch"
```

Comment:
just curious, where are the aarch64 platforms (tx1, tx2, hikey ...)?

Reply:
I've just translated into this "structure" what was used in the CAmke code before - which was `KernelSel4Arch`. For RISC-V, we don't have anything yet, because there was no need. We can do further refactoring based on this now, but I'd prefer to merge this first to have a baseline.

Reply:
@yyshen: again, I just translated what existed there before. Just not with the cleaner approach is becomes more obvious what happens here and what does not happen here. See the comment about this structure, there is says
```
If there was no match for any board the function will set:
     KernelPlatform=${PLATFORM}
 and leave all further setup to the architecture/platform specific
 configuration to <sel4 kernel>/src/plat/*/config.cmake
```
So what will happen is, that these file than act upon the setting of KernelPlatform. Either they have only one platform anyway or they have a fallback to a default board then. 
The whole things with the platform is still a bit of a mess and I think `correct_platform_strings()` only exists as an attempt to clean this up and address some issues that are eventually to be fixed in the `<sel4 kernel>/src/plat/*/config.cmake` files.  With my cleanup, it should get easier to get this started, as it becomes more clear now, what actually happens.

Reply:
This was originally done to avoid having to propagate platform name changes through all of the CI scripts that would refer to platforms by different names, such as rpi and bcm2837.  I think the inspiration of making aliases came from seeing what netbsd does: https://github.com/IIJ-NetBSD/netbsd-src/blob/master/build.sh#L618